distributed/diagnostics/plugin.py

```diff
@@ -282,34 +278,74 @@ def __init__(self, packages, pip_options=None, restart=False):
         self.packages = packages
         self.restart = restart
         self.pip_options = pip_options or []
+        self.id = f"pip-install-{uuid.uuid4()}"
```

Comment:
I think we should use the `worker.id` instead. I _think_ that there is a 1to1 relationship between PipInstallPlugin and Worker instances but either way, the uniqueness property should link to the worker, shouldn't ti?

Reply:
If I understand correctly, `worker.id` is unique _per worker instance_, i.e., we get a new `worker.id` upon every restart: 

https://github.com/dask/distributed/blob/aaab17c3190b5a2978832063e4b3923cc7c2ebbc/distributed/core.py#L341

This would cause a restart loop that the plugin never recovers from. A combination of the `socket.gethostname()` and `os.getpid()` might be even more accurate.

Reply:
See `PipInstall._compose_{installed|restarted}_key()` for the namespacing. We achieve the uniqueness per worker of `PipInstall._compose_restarted_key()` using `worker.nanny`, which _should_ be the correct key, but we may want to add the `pid` to that.

Reply:
worker.nanny is not perfect but I believe good enough. I don't think we have the PID of the nanny and I don't think this is necessary. Let's not overdo it here

Reply:
TIL: We teardown plugins that are overwritten which allows us to nicely remove their metadata. We can register also plugins to custom names ignoring their name attribute. However, we do not tell those plugins the name they are registered by. For `PipInstall` plugins, `plugin.name` will remain `pip`. We also do not pass that information on setup or teardown. Otherwise, we would be able to drop the ID and just use their name.